src/particles/particle_types.jl

```diff
@@ -0,0 +1,185 @@
+###############
+# The particle types 
+#
+# In this file, we define the types of particles, their states and directions, and
+# implement the abstact particle interface accordingly. 
+###############
+
+"""
+    AbstractParticleType <: AbstractParticle
+
+This is the abstract base type for every species of particles. All functionalities defined on subtypes of `AbstractParticleType` should be static, i.e. known at compile time. 
+For adding runtime information, e.g. four-momenta or particle states, to a particle, consider implementing a concrete subtype of [`AbstractParticle`](@ref) instead, which may have a type parameter `P<:AbstractParticleType`.
+
+Concrete built-in subtypes of `AbstractParticleType` are [`Electron`](@ref), [`Positron`](@ref) and [`Photon`](@ref).
+"""
+abstract type AbstractParticleType <: AbstractParticle end
+
+"""
+Abstract base types for particle species that act like fermions in the sense of particle statistics.
+
+!!! note "particle interface"
+    Every concrete subtype of [`FermionLike`](@ref) has `is_fermion(::FermionLike) = true`.
+"""
+abstract type FermionLike <: AbstractParticleType end
+
+is_fermion(::FermionLike) = true
+
+"""
+Abstract base type for fermions as distinct from [`AntiFermion`](@ref)s.
+    
+!!! note "particle interface"
```

Comment:
Again, is it necessary to duplicate the actual implementation here?

Reply:
I think these comments are somewhat helpful, it's just that yes, the implementations are so short that the documentation will be equal to the implementation.
Thoughts @szabo137 ?